invenio_records_marc21/system/service.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+#
+# Copyright (C) 2020-2021 CERN.
+# Copyright (C) 2020 Northwestern University.
+# Copyright (C) 2021 TU Wien.
+# Copyright (C) 2021 Graz University of Technology.
+#
+# Invenio-Records-Marc21 is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Marc21 Record Resource."""
+
+
+from flask_resources import route
+
+from . import config
+from .models import Marc21Templates
+
+
+#
+# Records
+#
+class Marc21TemplateService:
+    """Marc21 template resource."""
+
+    config_name = "MARC21_TEMPLATE_CONFIG"
+    default_config = config.Marc21TemplateConfig
+    template_cls = Marc21Templates
+
+    def __init__(self, config):
+        """Constructor for Marc21TemplateService."""
+        self.config = config
+
+    def create(self, name, data):
+        """Create a template for a new record."""
+        return self._create(self.template_cls, name, data)
+
+    def _create(self, template_cls, name, data):
+        template = template_cls.create(name=name, data=data)
+        return template
+
+    def get_templates(self, names=None, with_deleted=False):
+        """Get templates for a new record."""
+        return self.template_cls.get_templates(names=names, with_deleted=with_deleted)
+
+    def get_template(self, name):
+        """Get a template for a new record."""
+        return self.template_cls.get_template(name=name)
+
+    def delete(self, all, force, name):
+        """Create a template for a new record."""
+        if all:
+            return self._deleteall(self.template_cls, force=force)
```

Comment:
not necessary to break those functionalities into separated small functions. The _deleteall method is also only calling the self.template_cls.deleteall method. if it would be done more it is ok to have a separated function/method but in this case it is to much.